oeps/oep-0026-arch-xapi-integration.rst

```diff
@@ -0,0 +1,483 @@
+and paired with a short display string. It denotes the action that was performed by the
+Actor on the Object in the statement. We will use only values found in the registry and
+document which values are used for which edX events in the `Open edX Events`_ section below.
+
+The registry is automatically created from multiple profiles. For now, we will limit
+ourselves to only URIs prefixed by the following domains, in the following priority
+order (in case of conflicting names):
+
+* http://adlnet.gov
+* http://w3id.org
+* http://id.tincanapi.com
+
+If, by any chance, a verb needed by Open edX does not exist in the registry, then we
+will create a pull request to recommend adding it to the `central github repository of
+xAPI Profiles`_. 
+
+Here is an example of a **Verb** JSON value that we would generate:
+
+::
+
+    "verb": {
+        "id": "http://adlnet.gov/expapi/verbs/answered",
+        "display": { 
+            "en-US": "answered"
+        }
+    }
+
+
+See `Experience API Deep Dive: Verbs`_ for more information on xAPI Verbs.
+
+.. _xAPI registry: http://xapi.vocab.pub/verbs/index.html
+.. _central github repository of xAPI Profiles: https://github.com/adlnet/xapi-authored-profiles
+.. _`Experience API Deep Dive: Verbs`: https://xapi.com/deep-dive-verb/
+
+xAPI Object
+~~~~~~~~~~~
+
+Initially, the **Object** in an Open edX xAPI event will be an xAPI *Activity*, which is
+uniquely defined by a URI. (In the future, we may expand Objects to also be Actors in case
+of *social interactions*, and Statements in case of *voiding*.)
+
+The **id** field is a unique identifier. The `Open edX Events`_ section has specifics on
+which Open edX identifier is used in each event.
+
+The **type** of the Activity and the **name** of the Activity are included in the
+**definition** field.  Similar to Verbs, the type of the Activity is a standard URI
+taken from the `xAPI registry`_.
+
+Here is an example of an **Object** JSON value that we would generate:
+
+::
+
+    "object": {
+        "id": "https://courses.openedx.org/xblock/block-v1:openedx+origami-folding+1T2018+type@problem+block@abcd",
+        "definition": {
+            "type": "http://adlnet.gov/expapi/activities/question",
+            "name": { 
+                "en-US": "Question on mountain fold needed to create an origami crane base",
+            }
+        }
+    }
+
+See `Deep Dive: Object`_ for more information on xAPI Object.
+
+.. _`Deep Dive: Object`: https://xapi.com/deep-dive-object/
+
+xAPI Context
+~~~~~~~~~~~~
+
+The **Context** field allows us to embed additional contextual information in each statement.
+See specifics in the `Open edX Events`_ section below since it varies by event type.
+
+Here is an example of a **Context** JSON value that we would generate:
+
+::
+
+    "context": {
+        "registration": "openedx.org/enrollments/enrollment-v1:<anonymized-enrollment-id>",
+        "contextActivities": {
+            “parent”: [{
```

Comment:
One of the challenges with the parent is the array. analyze array is compute intensive. We keep the most important data in the extension. we want to keep in the parent the actual parent of the object (the unit or the sequence) but it will be hard to extract that later.

Reply:
So in which field do you send the "course" information for the event?  Did you take a look at the specific event types in the [spreadsheet](https://docs.google.com/spreadsheets/d/1Qx-1NkpCHXkWh8AagwHD5vzyBCdRXQVof10Ent_EDms/edit#gid=52329066)?  Currently, we proposed sending only 1 value in the parent - rather than a list.  But the parent is typically either the course or the parent module.